### PR TITLE
[4.0] Implementing administrator template preview

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/28932

### Summary of Changes
As title says. 

### Testing Instructions
Enable Preview in Template Options
In each template manager filter by Administrator client

### After patch

Templates Manager will show Preview button for Atum
Screen Shot 2020-05-05 at 08 23 33

Templates styles Manager will also get a Preview button
Screen Shot 2020-05-05 at 10 40 41

### Clicking Preview button
We now get
Screen Shot 2020-05-05 at 08 26 20

### CODE REVIEW

**Comment by SharkyKZ** [comment]
Can you remove the duplicated code? There is no need for client check anymore except when building the preview link.

**Comment by infograf768** [comment]
On it.

**Comment by infograf768** [comment]
@SharkyKZ Done

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 1943fb8a760a0f581c0082e5d18c80c7651fb9c9<br /><br />It does what the screenshots show. It seems testing helps learning - I did not know there was template positions preview!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28936">issues.joomla.org/tracker/joomla-cms/28936</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 1943fb8a760a0f581c0082e5d18c80c7651fb9c9<br /><br />Works as expected. Even duplicated and confirmed that other functions within the administrator templates and styles works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28936">issues.joomla.org/tracker/joomla-cms/28936</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28936">issues.joomla.org/tracker/joomla-cms/28936</a>.</sub>

File: administrator/components/com_templates/tmpl/styles/default.php [hunk 1]
```diff
@@ -42,11 +42,9 @@
 								<th scope="col">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_TEMPLATES_HEADING_STYLE', 'a.title', $listDirn, $listOrder); ?>
 								</th>
-								<?php if ($clientId === 0) : ?>
 									<th scope="col" style="width:5%" class="text-center">
 										<?php echo Text::_('COM_TEMPLATES_TEMPLATE_PREVIEW'); ?>
 									</th>
```
Comment (Quy) [inline administrator/components/com_templates/tmpl/styles/default.php]:
```suggestion
								<th scope="col" style="width:5%" class="text-center">
									<?php echo Text::_('COM_TEMPLATES_TEMPLATE_PREVIEW'); ?>
								</th>
```

Comment (infograf768) [reply administrator/components/com_templates/tmpl/styles/default.php]: tks

**Comment by infograf768** [comment]
Was only cs. Can stay rtc

**Comment by SharkyKZ** [comment]
Template preview doesn't show on PDO. `$item->client_id` needs to be cast to integer. But it was same before this patch, so can be fixed in separate PR.

**Comment by Quy** [comment]
Please fix conflicts.

**Comment by infograf768** [comment]
Should be OK now.

File: administrator/components/com_templates/tmpl/styles/default.php [hunk 2]
```diff
@@ -81,19 +79,21 @@
...
 									<?php endif; ?>
 								</th>
-								<?php if ($clientId === 0) : ?>
-									<td class="text-center">
-										<?php if ($this->preview && $item->client_id == '0') : ?>
+								<td class="text-center">
+									<?php if ($this->preview) : ?>
+										<?php if ($clientId === 0 && $item->client_id == '0') : ?>
 											<a target="_blank" href="<?php echo Uri::root() . 'index.php?tp=1&templateStyle=' . (int) $item->id ?>" class="jgrid">
+										<?php elseif ($clientId === 1 && $item->client_id == '1') : ?>
```
Comment (Quy) [inline administrator/components/com_templates/tmpl/styles/default.php:86]: Should this be `else` since it is either `0` or `1`?

**Pull request merged**